packages/editor/src/components/writing-flow/index.js

```diff
@@ -231,7 +231,11 @@ class WritingFlow extends Component {
 				}
 
 				// Set in case the meta key doesn't get released.
-				this.isEntirelySelected = isEntirelySelected( target );
+				// TinyMCE makes sure all content of the editable is select,
+				// but on some browsers (safari and firefox) calling isEntirelySelected right way still returns false.
+				// So to make sure we are compatible with this browsers we set is entirely selected to true
+				// after the first meta+a keypress assuming TinyMCE will make all content entirely selected.
+				this.isEntirelySelected = target.isContentEditable ? true : isEntirelySelected( target );
```

Comment:
I'm trying to wrap my head around this logic. When would primary+a **not** result in all of the text being selected? In other words, why can't this line be:

```js
this.isEntirelySelected = true;
```

Reply:
I'm very hesitant of anything which forks logic based on browser differences. Can we simplify to what @noisysocks said? Should that live in `isEntirelySelected` ? Do we need `isEntirelySelected` at all? What is it about being `isContentEditable` that makes us assume it's entirely selected?